src/CodeGeneratorGo.cs

```diff
@@ -36,6 +36,19 @@ public override string ImplementationFileExtension
         /// <returns></returns>
         public override async Task Generate(CodeModel cm)
         {
+            // if preview-chk:true is specified verify that preview swagger is output under a preview subdirectory.
+            // this is a bit of a hack until we have proper support for this in the swagger->sdk bot so it's opt-in.
+            if (Settings.Instance.Host.GetValue<bool>("preview-chk").Result)
+            {
+                const string previewSubdir = "/preview/";
+                var files = await Settings.Instance.Host.GetValue<string[]>("input-file");
+                // skip validation for composite builds as we don't have a well-defined model for mixed preview/stable swaggers
+                if (files.Length == 1 && files[0].IndexOf(previewSubdir) > -1 &&

```

Comment:
I would say if anything is preview than it should be in preview ? but sure we can leave it out for now

Reply:
I tend to agree, once we as a team can decide then we can update this.